includes/class-wcs-action-scheduler-customer-notifications.php

```diff
@@ -0,0 +1,340 @@
+<?php
+
+/**
+ * Scheduler for subscription events that uses the Action Scheduler
+ *
+ * @class     WCS_Action_Scheduler
+ * @version   1.0.0 - Migrated from WooCommerce Subscriptions v2.0.0
+ * @package   WooCommerce Subscriptions/Classes
+ * @category  Class
+ */
+class WCS_Action_Scheduler_Customer_Notifications extends WCS_Scheduler {
```

Comment:
While testing this, I discovered that when a user switches the payment type (manual/auto) from their My Account page, the scheduler does not recognize the change. See:
![Markup 2024-09-04 at 16 28 25](https://github.com/user-attachments/assets/da38a611-e821-4b55-839e-4868741af6f1)

This is likely unnecessary for the subscription core since it uses a single action for both types.

I think we could modify the constructor here to hook into the `woocommerce_subscription_object_updated_props` and reschedule when the `requires_manual_renewal` prop changes.

What do you think?

Reply:
To really close the loop, I changed the notification scheduling to only schedule one payment notification that decides which email to send based on subscription type at the time of notification.